cmd/esc/cli/env_set.go

```diff
@@ -71,6 +73,14 @@ func newEnvSetCmd(env *envCommand) *cobra.Command {
 				return fmt.Errorf("value looks like a secret; rerun with --secret to mark it as such, or --plaintext if you meant to leave it as plaintext")
 			}
 			if secret {
+				if yamlValue.Kind == yaml.ScalarNode && yamlValue.Tag != "!!str" {
+					err = yaml.Unmarshal([]byte(strconv.Quote(args[1])), &yamlValue)
+					if err != nil {
+						return errors.New("secret values must be strings; try surrounding the argument in both single and double quotes (e.g. '\"foo\"')")
```

Comment:
This error message is confusing to me since the point of the PR is to coerce non-string values into strings. It looks like the test case for this is `[]`, which you could quote, but seems like a type error. I'm not sure what to suggest instead.

Reply:
So this error message is a bit of paranoia on my part.

We only hit this branch if the value the user provided deserializes as a scalar, so it doesn't trip on arrays or objects. So we'd never reach here on `[]`. This guards against the transformation being performed by `strconv.Quote` going wrong in some way, and points the user towards a potential solution.

We could instead issue an internal error and direct the user to file a bug.

Reply:
thoughts @lunaris @lblackstone ?

Reply:
> We could instead issue an internal error and direct the user to file a bug.

I like this suggestion better. Ideally, we could print the error message from `yaml.Unmarshal`, but would want to avoid printing out the plaintext secret, so something more generic would be fine as well.

Reply:
Done